comp/metadata/packagesigning/packagesigningimpl/aptsigning.go

```diff
@@ -0,0 +1,192 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package packagesigningimpl
+
+import (
+	"bufio"
+	"encoding/xml"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	pkgUtils "github.com/DataDog/datadog-agent/comp/metadata/packagesigning/utils"
+)
+
+const (
+	trustedFolder  = "/etc/apt/trusted.gpg.d/"
+	trustedFile    = "/etc/apt/trusted.gpg"
+	mainSourceList = "/etc/apt/sources.list"
+	sourceList     = "/etc/apt/sources.list.d/"
+)
+
+var (
+	sourceListRegexp = regexp.MustCompile(`^([^\s]+)\s?(\[.*\]\s)?(.*)$`)
+	signedBy         = regexp.MustCompile(`signed-by=([A-Za-z0-9_\-\.\/]+)`)
+	trusted          = regexp.MustCompile(`trusted=yes`)
+	debsigPolicies   = "/etc/debsig/policies/"
+	debsigKeyring    = "/usr/share/debsig/keyrings/"
+)
+
+// getAPTSignatureKeys returns the list of debian signature keys
+func getAPTSignatureKeys(client *http.Client, logger log.Component) []signingKey {
+	cacheKeys := make(map[string]signingKey)
+	// debian 11 and ubuntu 22.04 will be the last using legacy trusted.gpg.d folder and trusted.gpg file
+	updateWithTrustedKeys(cacheKeys, client, logger)
+	// Regular files are referenced in the sources.list file by signed-by=filename
+	updateWithSignedByKeys(cacheKeys, client, logger)
+	// In APT we can also sign packages with debsig
+	keyPaths := getDebsigKeyPaths()
+	for _, keyPath := range keyPaths {
+		decryptGPGFile(cacheKeys, repoFile{filename: keyPath, repositories: nil}, "debsig", client, logger)
```

Comment:
Naming nit - I'm guessing we're not really decrypting the GPG files but rather decoding them.
```suggestion
		decryptGPGFile(cacheKeys, repoFile{filename: keyPath, repositories: nil}, "debsig", client, logger)
```

Similar comment for other use of `decrypt*` in this and other files

Reply:
I moved it to `readGPGFile` as I was not sure about decrypting or decoding :D